### ISSUE TITLE
503 Service Unavailable RPC::DEADLINE_EXCEEDED using java analytics api

### ISSUE THREAD

**Issue by liamsharp** [googleapis/google-api-java-client-services#6282]
We're using the v3-rev20190807-1.30.1 of com.google.apis google-api-services-analytics in our Java application. Our integration has been working fine for years, and still is, apart from for this one account. We are getting the following:

```
[{
  "error": {
    "code": 503,
    "message": "",
    "errors": [
      {
        "message": "",
        "domain": "global",
        "reason": "backendError",
        "debugInfo": "detail: \"[ORIGINAL ERROR] RPC::DEADLINE_EXCEEDED: Server deadline (19.99883673875s) expired.\"\n"
      }
    ],
    "status": "UNAVAILABLE"
  }
}
]
```

When I use the Query Explorer I don't see this response, however the requests do take over 20 seconds to complete, normally around 30 seconds.

Is there anyway around this issue? I can only find this page:

https://cloud.google.com/apis/design/errors

This suggests I might have control over this timeout, but I can't see anything in the API that allows me to control this.

**Comment by chingor13** [comment]
It looks like this is happening on the backend, but you may try upping the timeouts from the client: https://googleapis.github.io/google-api-java-client/errors.html#setting-timeouts

**Comment by chingor13** [comment]
Also, if you're interested, there's [an alpha gRPC client](https://github.com/googleapis/java-analytics-data) (vs. the http/json version published from here).

**Comment by liamsharp** [comment]
@chingor13 thanks for the response - yes, I did try that at the time, setting both timeouts to 2 mins, but this timeout happens after about 20 seconds. Like you say, it seems like a back end thing. We'll be looking at the client at the start of our next release, but that won't be for a couple of months.

It's a shame we can't get hold of anyone from the backend team to offer their thoughts on this. I did raise an issue here https://issuetracker.google.com/issues/170406297 also but that has gone quiet I'm afraid.
